R/tm_g_distribution.R

```diff
@@ -525,8 +522,8 @@ srv_distribution <- function(id,
               )
             }
 
-            ANL <- merged$anl_q_r()[[as.character(dist_var[[1]]$dataname)]]
-            round(get_dist_params(as.numeric(stats::na.omit(ANL[[dist_var2]])), input$t_dist), 2)
+            ANL <- merged$anl_q_r()[["ANL"]]
+            round(get_dist_params(as.numeric(stats::na.omit(ANL[[selector_list()$dist_i()$select]])), input$t_dist), 2)
```

Comment:
ANL variable name needs to be taken from `merge_expression_srv` values. In this module `merge_vars()` keeps these variables. This is needed when using multiple data-extracts from multiple datasets where selected column names are duplicated - then ANL output of the merge operation will have duplicated columns prefixed. Imagine selecting column `AGE` from `ADSL` and `AGE` from `ADVS` - then merged `ANL` will have two `AGE` columns and referring to `AGE` will throw an error - `merge_expressoion_srv` (for example `merged$anl_input_r()$columns_source$dist_i`) keeps track of the potential prefixing.

```suggestion
            round(get_dist_params(as.numeric(stats::na.omit(ANL[[merge_vars()$dist_var]])), input$t_dist), 2)
```